sdk/interpolate/interpolate.go

```diff
@@ -125,9 +125,12 @@ func Do(input string, vars map[string]string) (string, error) {
 					}
 				}
 
-				var defaultIsUsed bool
+				var isHandlingUnknownVars bool
```

Comment:
you can keep defaultIsUsed here, no need to add a special case for ternay

Reply:
It was mainly to avoid this syntax when the assert was not set

` "{{.assert  | ternary .foo .bar}}"` instead of typing `"{{.assert | default false | ternary .foo .bar}}"`

Reply:
@louisquentinjoucla  ok, can you add a unit test about this use case  (.assert not defined) and a perhaps a comment in code to explains why there is this special case please?

Reply:
@yesnault  Done!